### PR TITLE
Add OpenSearch as a full text search

### PR DESCRIPTION
**Pull request by luciferlinx101**
### Description

Add OpenSearch as a full text search for Core Loop V2 use case.

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update

### Checklist
- [x] My pull request is atomic and focuses on a single change.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have added the required tests.

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -377,6 +380,49 @@ func main() {
 	}
 	fmt.Println("WorkspaceGateway provided")
 
+	// Provide OpenSearch Client
+	err = c.Provide(func() (*opensearch.Client, error) {
```
Comment (czezi) [inline server.go:384]: We don't keep the creation of the client here in server.go, move to config module.

File: server.go [hunk 2]
```diff
@@ -377,6 +380,49 @@ func main() {
...
+			return nil, fmt.Errorf("error creating the OpenSearch client: %w", err)
+		}
+		return openSearchClient, nil
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	// Provide OpenSearch Repository
+	err = c.Provide(func(client *opensearch.Client, logger *zap.Logger) repository.CodeBaseSearchRepository {
```
Comment (czezi) [inline server.go:416]: Simplify provide

**Review by czezi — CHANGES_REQUESTED** [review]

File: server.go (hunk 2 above)
Comment (luciferlinx101) [reply server.go:389]: This will be needed as which implementation of the repository has to be injected has to be decided

File: server.go (hunk 1 above)
Comment (luciferlinx101) [reply server.go]: Yes moving it there

File: server.go (hunk 2 above)
Comment (czezi) [reply server.go:389]: "Will be", not needed now.

**Review by czezi — CHANGES_REQUESTED** [review]

File: app/services/code_base_full_text_search_service.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package services
+
+import (
+	"ai-developer/app/repositories"
+	"context"
+)
+
+type CodeBaseSearchService struct {
+	codeFullTextSearchRepository *repositories.CodeBaseOpenSearchRepository
```
Comment (czezi) [inline app/services/code_base_full_text_search_service.go:9]: This has to be CodeBaseSearchRepository

**Review by czezi — CHANGES_REQUESTED** [review]

File: app/services/code_base_full_text_search_service.go (hunk 3 above)
Comment (luciferlinx101) [reply app/services/code_base_full_text_search_service.go:9]: was using that only initially if you see my previous commit as we had to move away from 
`// Provide OpenSearch Repository
	err = c.Provide(func(client *opensearch.Client, logger *zap.Logger) repository.CodeBaseSearchRepository {`
	
	
	this was providing CodeBaseSearchRepository this as a particular implementation

Comment (luciferlinx101) [reply app/services/code_base_full_text_search_service.go:9]: https://github.com/TransformerOptimus/SuperCoder/pull/76#discussion_r1680820194 Referring this

**Comment by luciferlinx101** [comment]
Closing the PR as the approach has changed can get it back later if needed!

**Pull request closed without merging**